annet/gen.py

```diff
@@ -232,11 +233,30 @@ def _old_new_per_device(ctx: OldNewDeviceContext, device: Device, filterer: Filt
 
         entire_results = res.entire_results
         json_fragment_results = res.json_fragment_results
+        old_json_fragment_files = old_files.json_fragment_files
+
         new_files = res.new_files()
-        new_json_fragment_files = res.new_json_fragment_files(old_files.json_fragment_files)
+        new_json_fragment_files = res.new_json_fragment_files(old_json_fragment_files)
+
         if ctx.args.acl_safe:
             safe_new_files = res.new_files(safe=True)
 
+        if not ctx.args.no_acl:
```

Comment:
Just spotted: the `--no-acl` option is related to `Generator.acl()`, not to `--filter-acl`, and it should not be checked. This use case should mirror the behaviour of `build_filter_acl()/apply_acl()`.